packages/client/lib/CredentialRequestClient.ts

```diff
@@ -92,45 +76,66 @@ export class CredentialRequestClient {
     const { proofInput } = opts;
     const formatSelection = opts.format ?? this.credentialRequestOpts.format;
 
-    let format: OID4VCICredentialFormat = formatSelection as OID4VCICredentialFormat;
-    if (opts.version < OpenId4VCIVersion.VER_1_0_11) {
-      if (formatSelection === 'jwt_vc' || formatSelection === 'jwt') {
-        format = 'jwt_vc_json';
-      } else if (formatSelection === 'ldp_vc' || formatSelection === 'ldp') {
-        format = 'jwt_vc_json-ld';
-      }
-    }
-
-    if (!format) {
+    if (!formatSelection) {
       throw Error(`Format of credential to be issued is missing`);
-    } else if (format !== 'jwt_vc_json-ld' && format !== 'jwt_vc_json' && format !== 'ldp_vc') {
-      throw Error(`Invalid format of credential to be issued: ${format}`);
     }
+    const format = getUniformFormat(formatSelection);
     const typesSelection =
       opts?.credentialTypes && (typeof opts.credentialTypes === 'string' || opts.credentialTypes.length > 0)
         ? opts.credentialTypes
         : this.credentialRequestOpts.credentialTypes;
     const types = Array.isArray(typesSelection) ? typesSelection : [typesSelection];
     if (types.length === 0) {
       throw Error(`Credential type(s) need to be provided`);
-    } else if (!this.isV11OrHigher() && types.length !== 1) {
+    }
+    // FIXME: this is mixing up the type (as id) from v8/v9 and the types (from the vc.type) from v11
```

Comment:
Could you elaborate a bit more on the FIXME. AFAIK current logic is following the different spec versions as it should (to be clear I didn't check explicitly when doing this review).

I haven't merged #54, because I believe it made some assumptions about 'inline credentials', which I wasn't too sure about.

IMO the approach should simply be to always resolve the offer to a credentials supported like object as it is also defined in the server metadata. So than we do not care whether it was an id, or some inline objects

Reply:
IMO merging current work and dropping support for some older versions is more important than this discussion and it will be mute anyway.

Reply:
> ould you elaborate a bit more on the FIXME. AFAIK current logic is following the different spec versions as it should (to be clear I didn't check explicitly when doing this review).

So this is as I understand it, but between version 8 and 9/11 there was a change in what type means. Version 8 had a type which pointed to the credentials_supported. In draft 9/11 the `types` was added which referred to the actual values of the `type` in the VC (so an array and starts with `VerifiableCredential`). 

These were being used interchangeably in this library, and it resulted in some cases being handled a bit weirdly.  Because the getTypes method was adding the string values from an offer `credentials` together with the inline offer `types` property. And in this case the string values were pointers to the `id` property in `credentials_supported`, while the `types` were the types that would be included in the issued VC. I think in a lot of examples the `id` of a credential matched one of the `types` that would be put in the credential (e.g. `UniversityCredential` being used as the `credentials_supported.id`, as well as the second `type` in vc (`type: [VerifiableCredential, UniserityCredential]`). 

Hope this makes sense